File: Dopamine.InformationModule/ViewModels/InformationAboutViewModel.vb
Imports Dopamine.Common.Services.Dialog
Imports Dopamine.Core.Utils
Imports Microsoft.Practices.Prism.Commands
Imports Microsoft.Practices.Prism.Mvvm
Imports Microsoft.Practices.Unity

Namespace ViewModels
    Public Class InformationAboutViewModel
        Inherits BindableBase

#Region "Variables"
        Private mContainer As IUnityContainer
        Private mDialogService As IDialogService
#End Region

#Region "Commands"
        Public Property ShowLicenseCommand As DelegateCommand
#End Region

#Region "Construction"
        Public Sub New(iContainer As IUnityContainer, iDialogService As IDialogService)

            Me.mContainer = iContainer
            Me.mDialogService = iDialogService

            Me.ShowLicenseCommand = New DelegateCommand(Sub()

                                                            Dim view = Me.mContainer.Resolve(Of InformationAboutLicense)()

                                                            Me.mDialogService.ShowCustomDialog(&HE73E,
                                                                       16,
                                                                       ResourceUtils.GetStringResource("Language_License"),
                                                                       view,
                                                                       400,
                                                                       0,
                                                                       False,
                                                                       False,
                                                                       ResourceUtils.GetStringResource("Language_Ok"),
                                                                       String.Empty,
                                                                       Nothing)
                                                        End Sub)
        End Sub
#End Region
    End Class
End Namespace

File: Dopamine.InformationModule/ViewModels/InformationAboutViewModel.cs
using Digimezzo.Utilities.IO;
using Digimezzo.Utilities.Packaging;
using Digimezzo.Utilities.Utils;
using Dopamine.Common.Base;
using Dopamine.Common.Services.Dialog;
using Dopamine.Core.ViewModels;
using Dopamine.InformationModule.Views;
using Microsoft.Practices.Unity;
using Prism.Commands;

namespace Dopamine.InformationModule.ViewModels
{
    public sealed class InformationAboutViewModel : InformationAboutViewModelBase
    {
        #region Variables
        private IUnityContainer container;
        private IDialogService dialogService;
        private Package package;
        #endregion

        #region Commands
        public DelegateCommand ShowLicenseCommand { get; set; }
        #endregion

        #region Properties
        public Package Package
        {
            get { return this.package; }
            set { SetProperty<Package>(ref this.package, value); }
        }
        
        public override Core.Packaging.ExternalComponent[] Components
        {
            get {
                return new ProductInformation().Components;
            }
        }

        #endregion

        #region Construction
        public InformationAboutViewModel(IUnityContainer container, IDialogService dialogService)
        {
            this.container = container;
            this.dialogService = dialogService;

            Configuration config;
#if DEBUG
            config = Configuration.Debug;
#else
		    config = Configuration.Release;
#endif

            this.Package = new Package(ProcessExecutable.Name(), ProcessExecutable.AssemblyVersion(), config);

            this.ShowLicenseCommand = new DelegateCommand(() =>
            {
                var view = this.container.Resolve<InformationAboutLicense>();

                this.dialogService.ShowCustomDialog(
                    0xe73e,
                    16,
                    ResourceUtils.GetStringResource("Language_License"),
                    view,
                    400,
                    0,
                    false,
                    true,
                    true,
                    false,
                    ResourceUtils.GetStringResource("Language_Ok"),
                    string.Empty,
                    null);
            });
        }
        #endregion
    }
}
